src/Http/Middleware/CsrfProtectionMiddleware.php

```diff
@@ -207,7 +207,7 @@ protected function _addTokenCookie($token, ServerRequest $request, Response $res
             '',
             (bool)$this->_config['secure'],
             (bool)$this->_config['httpOnly'],
-            $this->_config['samesite']
+            isset($this->_config['samesite']) ? $this->_config['samesite'] : $this->_defaultConfig['samesite']
```

Comment:
Shoudnt the constructor take care of this?

    $this->_config = $config + $this->_defaultConfig;

What is the issue of merging that here would require an additional ternary?

Reply:
Usually, the constructor works.

The problem occurs if...

- the app is upgraded from 3.9.2 or earlier, to 3.9.3 (or later)
- and route cache is not deleted after the upgrade

because the instance of the middleware is serialized in cache, and constructor is not called.

Reply:
That sounds like a deployment issue to me, as you need to bust the cache after each deployment.
But workarounds here in the code for this seem counterproductive.

Reply:
Yes, it's a deployment issue and the new code looks ugly, but I believe this change helps some application developers.

Reply:
This will be a maintenance issue in the future without context.

Reply:
I don't believe it helps any developers, as there are still 100 other issues that can occur when not following basic principle of cache clearing when doing a deploy.
So while this might by accident then solve one, the others remain unsolved.

Reply:
At least this will be one fewer problem :smile: